scripts/ingest.py

```diff
@@ -79,22 +86,105 @@ def process_one_docs(directory, folder_name):
         # If no argument exists or the yes is not True, then the
         # user permission is requested to call the API.
         if len(sys.argv) > 1 and yes:
-            call_openai_api(docs, folder_name)
+            call_openai_api(docs, folder_name, use_s3, s3_assume_role)
         else:
-            get_user_permission(docs, folder_name)
-
-
-    folder_counts = defaultdict(int)
-    folder_names = []
-    for dir_path in dir:
-        folder_name = os.path.basename(os.path.normpath(dir_path))
-        folder_counts[folder_name] += 1
-        if folder_counts[folder_name] > 1:
-            folder_name = f"{folder_name}_{folder_counts[folder_name]}"
-        folder_names.append(folder_name)
-
-    for directory, folder_name in zip(dir, folder_names):
-        process_one_docs(directory, folder_name)
+            get_user_permission(docs, folder_name, use_s3, s3_assume_role)
+
+    if not use_s3:
```

Comment:
Please review the code below and extract meaningful methods to avoid multi-screen functions.
I recommend reading a couple of Kent Beck's books.